### ISSUE TITLE
`defaultValue` is not a member of `GetInputPropsReturnType` for `uncontrolled` forms

### ISSUE THREAD

**Issue by snturk** [mantinedev/mantine#6086]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.8.0

### What package has an issue?

@mantine/form

### What framework do you use?

create-react-app (CRA)

### In which browsers you can reproduce the issue?

All

### Describe the bug

When using `useForm` hook with mode `uncontrolled`, the result of `getInputProps('fieldName')` includes `defaultValue` but the type `GetInputPropsReturnType` does not have that property.

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by EmReedCodes** [comment]
Running into the exact same issue. Did you find any solutions or work arounds?

**Comment by pjnicolas** [comment]
I'm getting a similar issue when using an uncontrolled form in Next.js.

When using the "controlled" form everything works as expected, but for the "uncontrolled" version the `form.getInputProps(...)` function returns a "key" property, which is not expected by typescript.

So this code:

```tsx
<TextInput
  label="Name"
  {...form.getInputProps('name')}
/>
```

Gives me this error:

```
Warning: A props object containing a "key" prop is being spread into JSX:
  let props = {key: someKey, className: ..., label: ..., placeholder: ..., withAsterisk: ..., onChange: ..., error: ..., defaultValue: ..., onFocus: ..., onBlur: ...};
  <@mantine/core/TextInput {...props} />
React keys must be passed directly to JSX without using spread:
  let props = {className: ..., label: ..., placeholder: ..., withAsterisk: ..., onChange: ..., error: ..., defaultValue: ..., onFocus: ..., onBlur: ...};
  <@mantine/core/TextInput key={someKey} {...props} />
    at WorkerPersonalData (webpack-internal:///(app-pages-browser)/./components/worker/WorkerPersonalData.tsx:18:11)
    at div
    at WorkerItem (webpack-internal:///(app-pages-browser)/./components/worker/WorkerItem.tsx:14:11)
    at HydrationBoundary (webpack-internal:///(app-pages-browser)/./node_modules/@tanstack/react-query/build/modern/HydrationBoundary.js:14:11)
    at Page (Server)
    at InnerLayoutRouter (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/layout-router.js:242:11)
    at RedirectErrorBoundary (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/client/components/redirect-boundary.js:74:9)

...
```

...and I need to fix it with this ugly cast:

```tsx
<TextInput
  label="Name"
  {...form.getInputProps('name')}
  key={(form.getInputProps('name') as any).key}
/>
```

**Comment by rtivital** [comment]
Fixed in [7.8.1](https://github.com/mantinedev/mantine/releases/tag/7.8.1)
